internal/kubernetes/drainSchedule_test.go

```diff
@@ -77,10 +78,42 @@ func TestDrainSchedules_Schedule(t *testing.T) {
 			// Deleting schedule
 			scheduler.DeleteSchedule(tt.node.Name)
 			// Check that node is no more scheduled for drain
-			hasSchedule, _, _ = scheduler.HasSchedule(tt.node.Name)
+			hasSchedule, _ = scheduler.HasSchedule(tt.node.Name)
 			if hasSchedule {
 				t.Errorf("Node %v should not been scheduled anymore", tt.node.Name)
 			}
 		})
 	}
 }
+
+type failDrainer struct {
+	NoopCordonDrainer
+}
+
+func (d *failDrainer) Drain(n *v1.Node) error { return errors.New("myerr") }
+
+func TestDrainSchedules_Schedule_Polling(t *testing.T) {
+	scheduler := NewDrainSchedules(&failDrainer{}, &record.FakeRecorder{}, 0, zap.NewNop())
+	node := &v1.Node{ObjectMeta: meta.ObjectMeta{Name: nodeName}}
+
+	when, err := scheduler.Schedule(node)
+	if err != nil {
+		t.Fatalf("DrainSchedules.Schedule() error = %v", err)
+	}
+
+	timeout := time.After(time.Until(when) + time.Minute)
+	for {
+		hasSchedule, failed := scheduler.HasSchedule(node.Name)
+		if !hasSchedule {
+			t.Fatalf("Missing schedule record for node %v", node.Name)
+		}
+		if failed {
+			break
+		}
+		select {
+		case <-time.After(time.Second):
+		case <-timeout:
```

Comment:
When do we hit the `timeout` condition vs. falling through via the initial case statement on 114?

Reply:
Line 114 `case <-time.After(time.Second)` acts only as a sleep to ensure that the test is not continuously consuming CPU cycles. Line 115 `case <-timeout` is a stop condition to prevent the test from hanging forever if for some reason/bug the scheduled function passed to `time.AfterFunc` was never called. I think some comments would also be nice here, working on it.